handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -53,12 +54,12 @@
      * @param mapping the mapping of domain name to {@link SslContext}
      */
     @SuppressWarnings("unchecked")
-    public SniHandler(DomainNameMapping<? extends SslContext> mapping) {
+    public SniHandler(Mapping<? super String, ? extends SslContext> mapping) {
```

Comment:
Perhaps we should not deprecate it because otherwise a user who uses DomainNameMapping as a Mapping will also get a deprecation warning.